eng/Directory.Build.Data.targets

```diff
@@ -175,15 +175,22 @@
            Text="When UseProjectReferenceToAzureClients=true all Azure.* references should be Project References, but the following are not [@(ShouldBeProjectReference)]" />
   </Target>
 
-  <Target Name="RunApiCompat" AfterTargets="CoreBuild" Condition="'$(ApiCompatVersion)' != '' and '$(DesignTimeBuild)' != 'true' and '$(SkipApiCompat)' != 'true'">
-    <MSBuild
-      Projects="$(MSBuildThisFileDirectory)/ApiCompat/ApiCompat.csproj"
-      Properties="TargetPackageName=$(PackageId);
-                  TargetPackageVersion=$(ApiCompatVersion);
-                  TargetOutputPath=$(IntermediateOutputPath);
-                  BaseIntermediateOutputPath=$(IntermediateOutputPath)\ApiCompat\"
-      Targets="ApiCompatVerification"
-     />
+  <ItemGroup Condition="'$(ApiCompatVersion)' != ''">
+    <PackageDownload Include="$(PackageId)" Version="[$(ApiCompatVersion)]" />
+    <PackageReference Include="Microsoft.DotNet.ApiCompat" />
+  </ItemGroup>
+ 
+  <Target Name="_ResolveResolvedMatchingContract" BeforeTargets="ValidateApiCompatForSrc" Condition="'$(ApiCompatVersion)' != ''">
+    <ItemGroup>
+      <_ReferencePathDirectories Include="@(ReferencePath -> '%(RootDir)%(Directory)')"  />
+      <ResolvedMatchingContract Include="$(NuGetPackageRoot)\$(PackageId)\$(ApiCompatVersion)\lib\$(TargetFramework)\$(TargetFileName)">
+        <DependencyPaths>@(_ReferencePathDirectories->Distinct(), ',')</DependencyPaths>
+      </ResolvedMatchingContract>
+    </ItemGroup>
+  </Target>
+
+  <Target Name="ApiCompatFinishedMessage" AfterTargets="ValidateApiCompatForSrc" Condition="'$(ApiCompatVersion)' != ''">
+     <Message Text="Ran ApiCompat against %(ResolvedMatchingContract.Identity) using assemblies from $(IntermediateOutputPath)" Importance="High" />
```

Comment:
1. I don't see it in this PR, but are still saying we're about to run API compat? We should, so you know - if that's hanging, failing (bombing out), etc. - probably what it was.
2. This doesn't actual fix the bug you tagged about failing the build, but there may be an opportunity here. The binlogs show a property is set that you could check here and use the `Error` task to fail the build and vary the message emitted. Then you could fix both bugs I opened about these related issues.

Reply:
I mostly put this as a way to help know that this is running. The exit works as it should and it should fail the build. I just submitted a test break to verify.

Reply:
Looks like it is working : https://dev.azure.com/azure-sdk/public/_build/results?buildId=542572&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=80694584-9a00-5514-2ba2-f6dabf815050&l=440